### PR TITLE
Bug 1207221 - Do not turn on the screen when receiving a notification during a call r=etienne

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ec34d25f7dec507b6e8df9c229141efbd2a1ba0f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=115217b3ab78237b969244dfa2a3afef4a83290f

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=0277122f092247c7256ad1a2d615849575996c6d

File: apps/system/js/notification_screen.js [hunk 1]
```diff
@@ -578,10 +586,7 @@ var NotificationScreen = {
           // bug 1050023: disable vibration for downloads when asleep
           if (type.indexOf('download-notification-downloading') === -1 &&
               manifestURL.indexOf('email.gaiamobile.org') === -1) {
-            window.addEventListener('visibilitychange', function waitOn() {
```
Comment (etiennesegonzac) [inline apps/system/js/notification_screen.js]: nit: we can probably make the logic a bit more readable now like
```js
var vibrateWhenAsleep = type.indexOf('download-notification-downloading') === -1 &&
                        manifestURL.indexOf('email.gaiamobile.org') === -1
if (vibrateWhenAsleep || !document.hidden) {
  navigator.vibrate(pattern);
}
```

File: apps/system/js/notification_screen.js [hunk 2]
```diff
@@ -509,9 +509,18 @@ var NotificationScreen = {
...
     // We turn the screen on if needed in order to let
-    // the user see the notification toaster
+    // the user see the notification toaster unless we're on a call
+    var telephony = window.navigator.mozTelephony;
+    var numOpenLines = 0;
+
+    if (telephony) {
+      numOpenLines = telephony.calls.length +
+        (telephony.conferenceGroup.calls.length ? 1 : 0);
+    }
```
Comment (etiennesegonzac) [inline apps/system/js/notification_screen.js]: Feels like this bit of logic is getting repeated all over the place :)

This is totally optional, or can be a follow up, but I think the `DialerAgent` should have an `onCall` method returning `numOpenLines > 0` and we should `Service.registerState('onCall', this)` when we start the `DialerAgent`.

This way we could simplify the `DialerAgent` a tiny bit and simply do a `Service.query('onCall')` here.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=8c258b3d689df9698e267901687b5cc3a467640d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e058d82fab89b887d31c18e53002c97368bd9c61

File: apps/system/js/dialer_agent.js [hunk 3]
```diff
@@ -254,5 +258,12 @@
     }
   };
 
+  DialerAgent.prototype.onCall = function() {
```
Comment (etiennesegonzac) [inline apps/system/js/dialer_agent.js]: nit: I think we have a `this._telephony`

Comment (etiennesegonzac) [inline apps/system/js/dialer_agent.js]: we have 2 `numOpenLines > 0` checks in `handleEvent` that should be replaced by this

Comment (etiennesegonzac) [inline apps/system/js/dialer_agent.js]: and one if `freeCalscreenWindow`

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bf35a66ad75d4a96e27c955dedd40ce19e3059e2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e3f1b0babf555be150dcdd3c715c435e035bba67

**Pull request merged**